### PR TITLE
colored sparkly

### LINKED ISSUE #8: Colored sparks
See https://github.com/holman/spark/pull/88.

### PR DESCRIPTION
**Pull request by beatfreaker**
@sindresorhus please check

-

Fixes #8.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,6 +13,10 @@ module.exports = function (numbers, opts) {
 	var finiteNumbers = numbers.filter(isFinite);
 	var min = opts.min || Math.min.apply(null, finiteNumbers);
 	var max = opts.max || Math.max.apply(null, finiteNumbers);
+	var color = opts.color || '';
+	if (color === 'fire') {
+		ticks = [colors.fg.getRgb(5,5,3) + '▁', colors.fg.getRgb(5,5,4) + '▂', colors.fg.getRgb(5,5,0) + '▃', colors.fg.getRgb(5,4,0) + '▄', colors.fg.getRgb(5,3,0) + '▅', colors.fg.getRgb(5,2,0) + '▆', colors.fg.getRgb(5,1,0) + '▇', colors.fg.getRgb(5,0,0) + '█'];
```
Comment (sindresorhus) [inline index.js]: You should `color.reset` at the end of each item.

**Comment by sindresorhus** [comment]
Tests are failing: https://travis-ci.org/sindresorhus/sparkly/jobs/95621077

File: index.js [hunk 2]
```diff
@@ -12,6 +13,10 @@ module.exports = function (numbers, opts) {
 	var finiteNumbers = numbers.filter(isFinite);
 	var min = opts.min || Math.min.apply(null, finiteNumbers);
 	var max = opts.max || Math.max.apply(null, finiteNumbers);
+	var color = opts.color || '';
```
Comment (sindresorhus) [inline index.js]: Call the option `style` instead of `color`.

**Comment by sindresorhus** [comment]
Can you also document the option in the readme and add a test?

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js]: I would also probably only define the colors in an array `[[5,5,3], [5,5,4], ...]` and then loop through it and apply the color function and the spark according to the index.

**Comment by beatfreaker** [comment]
@sindresorhus please check now.

File: readme.md [hunk 3]
```diff
@@ -29,8 +29,14 @@ sparkly([0, 3, 5, '', 4, 3, 4, 10]);
 // specifying a min max object will change the sparkline range
 sparkly([1, 2, 3, 4, 5], {min: 0, max: 10});
 //=> '▁▂▃▄▄'
+
+// specifying a style object will change the sparkline color
+sparkly([1, 2, 3, 4, 5], {style: 'fire'});
+//=>'▁▂▃▄▅▆▇█'
 ```
 
+colored sparkline will look like this: ![sparkly](http://i.imgur.com/5SUQ4mz.png)
```
Comment (SamVerschueren) [inline readme.md]: Add the image directly to the repository or like the logo, add to `cloud.githubusercontent`. Not sure how do that cloud stuff though.

File: test.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import test from 'ava';
 import fn from './';
+var colors = require('ansi-256-colors');
```
Comment (SamVerschueren) [inline test.js]: `import colors from 'ansi-256-colors';`

File: test.js [hunk 5]
```diff
@@ -25,3 +26,7 @@ test('min and max arguments set graph range', t => {
 	t.is(fn([10, 11, 12, 13], {min: 0}), '▇▇██');
 	t.is(fn([10, 20, 30, 40, 50], {max: 100}), '▁▂▃▄▄');
 });
+
+test('colored graph', t => {
+	t.is(fn([1, 2, 3, 4, 5, 6, 7, 8], {style: 'fire'}), colors.fg.getRgb(5, 5, 3) + '▁' + colors.fg.getRgb(5, 5, 4) + '▂' + colors.fg.getRgb(5, 5, 0) + '▃' + colors.fg.getRgb(5, 4, 0) + '▄' + colors.fg.getRgb(5, 3, 0) + '▅' + colors.fg.getRgb(5, 2, 0) + '▆' + colors.fg.getRgb(5, 1, 0) + '▇' + colors.fg.getRgb(5, 0, 0) + '█');
```
Comment (SamVerschueren) [inline test.js]: Test isn't very clean to read.

```js
test('colored graph', t => {
	const graph = [`${colors.fg.getRgb(5, 5, 3)}▁`, `${colors.fg.getRgb(5, 5, 4)}▃`, `${colors.fg.getRgb(5, 5, 0)}▃`, `${colors.fg.getRgb(5, 4, 0)}▄`, `${colors.fg.getRgb(5, 3, 0)}▅`, `${colors.fg.getRgb(5, 2, 0)}▆`, `${colors.fg.getRgb(5, 1, 0)}▇`, `${colors.fg.getRgb(5, 0, 0)}█`];

	t.is(fn([1, 2, 3, 4, 5, 6, 7, 8], {style: 'fire'}), graph.join(''));
});
```

Not sure wether this is that much better though, but you should definitely use [template strings](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/template_strings). Let @sindresorhus decide how he wants this :).

File: index.js [hunk 6]
```diff
@@ -12,6 +13,10 @@ module.exports = function (numbers, opts) {
 	var finiteNumbers = numbers.filter(isFinite);
 	var min = opts.min || Math.min.apply(null, finiteNumbers);
 	var max = opts.max || Math.max.apply(null, finiteNumbers);
+	var style = opts.style || '';
+	if (style === 'fire') {
+		ticks = [colors.fg.getRgb(5, 5, 3) + '▁', colors.fg.getRgb(5, 5, 4) + '▂', colors.fg.getRgb(5, 5, 0) + '▃', colors.fg.getRgb(5, 4, 0) + '▄', colors.fg.getRgb(5, 3, 0) + '▅', colors.fg.getRgb(5, 2, 0) + '▆', colors.fg.getRgb(5, 1, 0) + '▇', colors.fg.getRgb(5, 0, 0) + '█'];
```
Comment (SamVerschueren) [inline index.js]: I wouldn't copy the sparks all over again, just create a color array and add the colors in the map function below if `style` is set to `fire`.

```js
var fire = [colors.fg.getRgb(5, 5, 3), colors.fg.getRgb(5, 5, 4), colors.fg.getRgb(5, 5, 0), ...];

// ...

return (style === 'fire' ? fire[tickIndex] : '') + ticks[tickIndex];
```

File: test.js (hunk 5 above)
Comment (sindresorhus) [inline test.js]: No point in testing all the colors as you're just duplicating the main code. Testing the color of the first spark should be enough.

File: index.js [hunk 7]
```diff
@@ -12,6 +13,10 @@ module.exports = function (numbers, opts) {
 	var finiteNumbers = numbers.filter(isFinite);
 	var min = opts.min || Math.min.apply(null, finiteNumbers);
 	var max = opts.max || Math.max.apply(null, finiteNumbers);
+	var style = opts.style || '';
```
Comment (SamVerschueren) [inline index.js]: No need to create a separate `style` variable. Just use `opts.style === 'fire'`.

File: readme.md (hunk 3 above)
Comment (beatfreaker) [inline readme.md]: @sindresorhus can you help me in uploading image to `cloud.githubusercontent` like the logo.

Comment (sindresorhus) [inline readme.md]: Just ignore that for now. I'll do a proper screenshot after this PR ;)

**Comment by beatfreaker** [comment]
@sindresorhus please check now

**Comment by sindresorhus** [comment]
@beatfreaker I think you missed https://github.com/sindresorhus/sparkly/pull/9#discussion_r47031634

**Comment by beatfreaker** [comment]
@sindresorhus so i should do like following

```js
[[5,5,3], [5,5,4], ...]

then access it like this

return (opts.style === 'fire' ? colors.fg.getRgb(color[tickIndex][0], color[tickIndex][1], color[tickIndex][2]) : '') + ticks[tickIndex] + (opts.style === 'fire' ? colors.reset : '');
```

doing something like this is correct?

**Comment by SamVerschueren** [comment]
I would go for this

```js
var color = [[5,5,3], [5,5,4], ...];

if(opts.style === 'fire') {
    return `${colors.fg.getRgb.apply(colors.fg, color[tickIndex])}${ticks[tickIndex}${colors.reset}`;
}

return ticks[tickIndex];
```

Don't like the ternary operator if it gets more complex.

**Comment by sindresorhus** [comment]
:+1: what @SamVerschueren suggested.

**Comment by beatfreaker** [comment]
Thanks @SamVerschueren

**Comment by beatfreaker** [comment]
Thanks @sindresorhus , please check and update if require any update.

File: test.js [hunk 8]
```diff
@@ -25,3 +26,7 @@ test('min and max arguments set graph range', t => {
 	t.is(fn([10, 11, 12, 13], {min: 0}), '▇▇██');
 	t.is(fn([10, 20, 30, 40, 50], {max: 100}), '▁▂▃▄▄');
 });
+
+test('colored graph', t => {
+	t.is(fn([1], {style: 'fire'}), colors.fg.getRgb(5, 5, 3) + '▅' + colors.reset);
```
Comment (SamVerschueren) [inline test.js]:
```js
t.is(fn([1], {style: 'fire'}), `${colors.fg.getRgb(5, 5, 3)}▅${colors.reset}`);
```

**Comment by SamVerschueren** [comment]
Add it to the options in the documentation as well.

**Comment by sindresorhus** [comment]
Thank you @beatfreaker. Appreciate the PR :)

Would you be interested in adding it to https://github.com/sindresorhus/sparkly-cli too?

**Comment by beatfreaker** [comment]
@sindresorhus sure :+1:

**Pull request closed without merging**